### DISCUSSION TITLE
SolvePnp input parameters error

### DISCUSSION THREAD

**Discussion by tod0086** [emgucv/emgucv#615] [Q&A]
Hello,
I'm trying to use the SolvePnp function in a C# project but can't find the right parameter type to use.

This is a simplified version of the code I'm using:

Mat cameraMatrix;   
Mat distCoeffs;          
//cameraMatrix and distCoeffs calculation though camera calibration...
//...
Mat rotationVector = new Mat();
Mat translationVector = new Mat();
VectorOfPointF image_points = new VectorOfPointF();
//filling image_points...
VectorOfPoint3D32F object_points = new VectorOfPoint3D32F();
//filling object_points...

 CvInvoke.SolvePnP(object_points , image_points, cameraMatrix, distCoeffs, rotationVector, translationVector);

Error I'm getting:
Emgu.CV.Util.CvException: 'OpenCV: CV_IS_MAT(objectPoints) && CV_IS_MAT(imagePoints) && CV_IS_MAT(A) && CV_IS_MAT(rvec) && CV_IS_MAT(tvec)'

I can't find the right way to convert object_points and image_pointsin to the correct format.

**Comment by emgucv** [comment]
This unit test ( TestChessboardCalibrationSolvePnP ) shows how to use CvInvoke.SolvePnP:
https://github.com/emgucv/emgucv/blob/master/Emgu.CV.Test/AutoTestCalibration.cs#L115
